src/mmrelay/meshtastic_utils.py

```
@@ -956,14 +1533,21 @@ def _get_device_metadata(client: Any) -> dict[str, Any]:
+            result["firmware_version"] = cached_firmware
+            result["success"] = True
+            return result
+
         # Preflight: client may be a mock without localNode/getMetadata
-        if not getattr(client, "localNode", None) or not hasattr(
-            client.localNode, "getMetadata"
+        if not getattr(client, "localNode", None) or not callable(
+            getattr(client.localNode, "getMetadata", None)
         ):
+            if raise_on_error:
+                raise _missing_metadata_probe_error()
```

Ruff (0.15.4)

[warning] Abstract `raise` to an inner function

(TRY301)